core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerClientCallbacks.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A collection of callbacks that are invoked for each request by {@link CircuitBreakerClient}.
+ * Users may implement this class in conjunction with {@link CircuitBreakerClientHandler} to
+ * use arbitrary CircuitBreaker implementations with {@link CircuitBreakerClient}.
+ * See {@link CircuitBreakerClientHandler#request(ClientRequestContext, Request)}
+ * for more information.
+ */
+@UnstableApi
+public interface CircuitBreakerClientCallbacks {
```

Comment:
`ClientRequestContext` is used for the parameter of the callbacks.
Should we use `RequestContext` and let users cast the type?
It seems to find because, in most cases, the `ctx` would not be used for reporting.

Reply:
Done! ✅